### PR TITLE
contracts-bedrock: delete dead code

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Deletes dead code and simplifies the deploy script
infrastructure to make it less error prone and more
explicit.

- **contracts-bedrock: simplify L1 state dump**
- **contracts-bedrock: explicit deploy config**
- **contracts-bedrock: fix possible error**
- **config: cleanup**
- **deploy-config: fixup**
- **contracts-bedrock: delete name function**
- **contracts: delete dead code**

### CODE REVIEW

**Comment by tynes** [comment]
This PR exists to improve the devex of using the deploy script so that its always expected to provide the deploy config path as an env var when running the deploy script

File: packages/contracts-bedrock/scripts/Config.sol [hunk 1]
```diff
@@ -15,15 +14,18 @@ library Config {
...
+            "DEPLOYMENT_OUTFILE", string.concat(vm.projectRoot(), "/deployments/", vm.toString(block.chainid), "-deploy.json")
         );
     }
 
     /// @notice Returns the path on the local filesystem where the deploy config is
     function deployConfigPath() internal view returns (string memory _env) {
-        _env = vm.envOr(
-            "DEPLOY_CONFIG_PATH", string.concat(vm.projectRoot(), "/deploy-config/", _getDeploymentContext(), ".json")
-        );
+        if (vm.isContext(VmSafe.ForgeContext.Test)) {
```
Comment (tynes) [inline packages/contracts-bedrock/scripts/Config.sol]: I think this should be `devnetL1` and not `hardhat`

File: bedrock-devnet/devnet/__init__.py [hunk 2]
```diff
@@ -142,10 +142,12 @@ def devnet_l1_allocs(paths):
     init_devnet_l1_deploy_config(paths)
 
     fqn = 'scripts/Deploy.s.sol:Deploy'
-    # Use foundry pre-funded account #1 for the deployer
     run_command([
-        'forge', 'script', '--chain-id', '900', fqn, "--sig", "runWithStateDump()", "--private-key", "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
-    ], env={}, cwd=paths.contracts_bedrock_dir)
```
Comment (mds1) [inline bedrock-devnet/devnet/__init__.py:148]: I'm not too familiar with the devnet setup—who is the new deployer now?

File: packages/contracts-bedrock/scripts/Config.sol [hunk 3]
```diff
@@ -15,15 +14,19 @@ library Config {
...
+            string.concat(vm.projectRoot(), "/deployments/", vm.toString(block.chainid), "-deploy.json")
         );
     }
 
     /// @notice Returns the path on the local filesystem where the deploy config is
     function deployConfigPath() internal view returns (string memory _env) {
-        _env = vm.envOr(
-            "DEPLOY_CONFIG_PATH", string.concat(vm.projectRoot(), "/deploy-config/", _getDeploymentContext(), ".json")
-        );
+        if (vm.isContext(VmSafe.ForgeContext.Test) || vm.isContext(VmSafe.ForgeContext.Snapshot)) {
```
Comment (mds1) [inline packages/contracts-bedrock/scripts/Config.sol]: This way coverage is also included
```suggestion
        if (vm.isContext(VmSafe.ForgeContext.TestGroup)) {
```

File: packages/contracts-bedrock/scripts/Config.sol [hunk 4]
```diff
@@ -15,15 +14,19 @@ library Config {
...
+            _env = string.concat(vm.projectRoot(), "/deploy-config/hardhat.json");
```
Comment (mds1) [inline packages/contracts-bedrock/scripts/Config.sol:25]: Why do we still call this hardhat? Can/should it be renamed?

File: bedrock-devnet/devnet/__init__.py (hunk 2 above)
Comment (tynes) [reply bedrock-devnet/devnet/__init__.py:148]: The default foundry sender is used

File: packages/contracts-bedrock/scripts/Config.sol (hunk 4 above)
Comment (tynes) [reply packages/contracts-bedrock/scripts/Config.sol:25]: We should prob rename to testing.json, happy to do in this PR or a follow up

Comment (mds1) [reply packages/contracts-bedrock/scripts/Config.sol:25]: Let's do a follow up in case it's invasive

**Review by mds1 — APPROVED** [review]

**Review by trianglesphere — APPROVED** [review]

**Pull request merged**